addons/gamification/models/gamification_karma_rank.py

```diff
@@ -35,7 +35,7 @@ def _compute_rank_users_count(self):
     @api.model_create_multi
     def create(self, values_list):
         res = super(KarmaRank, self).create(values_list)
-        users = self.env['res.users'].sudo().search([('karma', '>', 0)])
+        users = self.env['res.users'].sudo().search([('karma', '>', min(res.mapped('karma_min'), default=0))])
         users._recompute_rank()
```

Comment:
Just to avoid useless computation

```suggestion
        if users:
            users._recompute_rank()
```